src/mscorlib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -430,24 +433,14 @@ public override void BeginCatchBlock(Type exceptionType)
 
                 // if this is a catch block the exception will be pushed on the stack and we need to update the stack info
                 UpdateStackSize(OpCodes.Nop, 1);
-            }
 
-            current.MarkCatchAddr(ILOffset, exceptionType);
+                current.MarkCatchAddr(ILOffset, exceptionType);
```

Comment:
Yes. What if peak stack size is inside the filter? If the `UpdateStackSize` here was covering both cases then `GetMaxStackSize` would return one less than it should.